packages/frontend/core/src/mixpanel/index.ts

```diff
@@ -1,7 +1,88 @@
-import type { PlanChangeStartedEvent } from './plan-change-started';
-import type { PlanChangeSucceededEvent } from './plan-change-succeed';
+import { DebugLogger } from '@affine/debug';
+import type { OverridedMixpanel } from 'mixpanel-browser';
+import mixpanelBrowser from 'mixpanel-browser';
 
-export interface MixpanelEvents {
-  PlanChangeStarted: PlanChangeStartedEvent;
-  PlanChangeSucceeded: PlanChangeSucceededEvent;
+import type { GeneralMixpanelEvent, MixpanelEvents } from './events';
+
+const logger = new DebugLogger('mixpanel');
+
+function createMixpanel() {
+  let mixpanel;
+  if (process.env.MIXPANEL_TOKEN) {
+    mixpanelBrowser.init(process.env.MIXPANEL_TOKEN || '', {
+      track_pageview: true,
+      persistence: 'localStorage',
+      api_host: 'https://telemetry.affine.run',
+    });
+    mixpanel = mixpanelBrowser;
+  } else {
+    mixpanel = new Proxy(
+      function () {} as unknown as OverridedMixpanel,
+      createProxyHandler()
+    );
+  }
+
+  const wrapped = {
+    reset() {
+      mixpanel.reset();
+      mixpanel.register({
+        appVersion: runtimeConfig.appVersion,
+        environment: runtimeConfig.appBuildType,
+        editorVersion: runtimeConfig.editorVersion,
+        isSelfHosted: Boolean(runtimeConfig.isSelfHosted),
+        isDesktop: environment.isDesktop,
+      });
+    },
+    track<
+      T extends string,
+      P extends (T extends keyof MixpanelEvents
+        ? MixpanelEvents[T]
+        : Record<string, unknown>) &
+        GeneralMixpanelEvent,
+    >(event_name: T, properties?: P) {
+      logger.debug('track', event_name, properties);
```

Comment:
not sure if it is a good idea to enable it on production

Reply:
why not

Reply:
If you worry about performance, there shouldn't be much performance hit if you don't have the browser devtools open.

Maybe we should disable debug output for production by default in @affine/debug so there are no issues, but let's just leave it here for now.

Reply:
IIRC I only enabled logging in the previous version because there was no way of debugging  mixpanel locally previously.